lib/matplotlib/colors.py

```diff
@@ -793,25 +795,23 @@ def __init__(self, colors, name='from_list', N=None):
 
             the list will be extended by repetition.
         """
-        self.colors = colors
         self.monochrome = False  # True only if all colors in map are
                                  # identical; needed for contouring.
         if N is None:
-            N = len(self.colors)
+            self.colors = colors
+            N = len(colors)
         else:
-            if isinstance(self.colors, six.string_types):
-                self.colors = [self.colors] * N
+            if isinstance(colors, six.string_types):
```

Comment:
yes, but i don't think this would make things more legible (N is None and N is not None are two logical alternatives)